interactive_engine/executor/engine/pegasus/pegasus/tests/timeout_test.rs

```diff
@@ -0,0 +1,49 @@
+//
+//! Copyright 2020 Alibaba Group Holding Limited.
+//!
+//! Licensed under the Apache License, Version 2.0 (the "License");
+//! you may not use this file except in compliance with the License.
+//! You may obtain a copy of the License at
+//!
+//! http://www.apache.org/licenses/LICENSE-2.0
+//!
+//! Unless required by applicable law or agreed to in writing, software
+//! distributed under the License is distributed on an "AS IS" BASIS,
+//! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//! See the License for the specific language governing permissions and
+//! limitations under the License.
+
+use pegasus::api::{IterCondition, Iteration, Map, Sink};
+use pegasus::JobConf;
+use std::time::Duration;
+
+/// test binary merge pipeline;
+#[test]
+fn timeout_test_01() {
+    let mut conf = JobConf::new("timeout_test_01");
+    conf.time_limit = 5000;
+    let mut result = pegasus::run(conf, || {
+        |input, output| {
+            input
+                .input_from(vec![0u32])?
+                .iterate_until(IterCondition::max_iters(20), |iter| {
```

Comment:
Can add another test of two workers: one worker cancel, the other all good. ?

Reply:
Done